src/LiveComponent/src/EventListener/LiveComponentSubscriber.php

```diff
@@ -194,21 +200,26 @@ public function onKernelController(ControllerEvent $event): void
      */
     private function parseDataFor(Request $request): array
     {
-        if ($request->query->has('data')) {
-            return [
-                'data' => json_decode($request->query->get('data'), true, 512, \JSON_THROW_ON_ERROR),
-                'args' => [],
-                'actions' => [],
-            ];
-        }
+        if (!$request->attributes->has('_live_request_data')) {
```

Comment:
Added this so that we could call this method ad-hoc as many times as we want (which helps make the above methods more clear) and not worry about performance.